### PR TITLE
implement Ruff

### PR DESCRIPTION
**Pull request by bandophahita**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -48,40 +63,37 @@ classifiers = [
...
 requests = ">=2.26.0"
-types-requests = "2.25.0"
+types-requests = ">=2.26.0"
 
 
 # convenience packages for development of screenpy_requests only
 black = {version = "*", optional = true}
 coverage = {version = "*", optional = true}
-flake8 = {version = "*", optional = true}
 isort = {version = "*", optional = true}
```
Comment (perrygoy) [inline pyproject.toml]: Do we want to remove `isort` now that we figured `ruff`'s version out?

File: screenpy_requests/questions/body_of_the_last_response.py [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-from screenpy import Actor
 from screenpy.exceptions import UnableToAnswer
 from screenpy.pacing import beat
 
 from ..abilities import MakeAPIRequests
 
+if TYPE_CHECKING:
+    from screenpy import Actor
+
 subscripts = Union[str, int, slice]
```
Comment (perrygoy) [inline screenpy_requests/questions/body_of_the_last_response.py]: We can move this (and the `from typing import Union`) into the `if TYPE_CHECKING` block.

File: screenpy_requests/questions/headers_of_the_last_response.py [hunk 3]
```diff
@@ -1,15 +1,17 @@
-"""
-Investigate the headers of the last API response received.
-"""
+"""Investigate the headers of the last API response received."""
 
-from typing import MutableMapping
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, MutableMapping
```
Comment (perrygoy) [inline screenpy_requests/questions/headers_of_the_last_response.py:5]: We can move the `MutableMapping` into the `if TYPE_CHECKING` block.

File: tests/test_actions.py [hunk 4]
```diff
@@ -82,12 +96,14 @@ def test_hides_secret_headers(self, APITester, caplog):
         assert str(test_headers) not in caplog.text
 
 
-def test_generate_send_method_class_docstring():
+def test_generate_send_method_class_docstring() -> None:
     """Generated class and method's docstring both contain method name."""
-    test_method = "TEST"
+    test_method: str = "TESTYTEST"
```
Comment (perrygoy) [inline tests/test_actions.py]: Was this change really necessary? Weird.

**Review by perrygoy — DISMISSED** [review]
Thank you for all this polish, @bandophahita! 

I had a few very minor suggestions that are not blocking.

File: pyproject.toml (hunk 1 above)
Comment (bandophahita) [reply pyproject.toml]: Good call.  Removed.

File: screenpy_requests/questions/body_of_the_last_response.py (hunk 2 above)
Comment (bandophahita) [reply screenpy_requests/questions/body_of_the_last_response.py]: Moved the alias. I let ruff automatically handle the imports.  
I noticed it doesn't bother moving the union down into the TYPE_CHECKING block when other items are being imported from the module.  My guess is it doesn't save anything.

File: screenpy_requests/questions/headers_of_the_last_response.py (hunk 3 above)
Comment (bandophahita) [reply screenpy_requests/questions/headers_of_the_last_response.py:5]: Same deal here as the Union.  I'm guessing ruff sees this as a no-op move so it opts to leave it alone.

File: tests/test_actions.py (hunk 4 above)
Comment (bandophahita) [reply tests/test_actions.py]: I opted to use an uncommon (guaranteed not to be present in the docstring) value; lowering the chance of a false positive.

Comment (perrygoy) [reply tests/test_actions.py]: Sorry, i should have been more specific—was adding `: str` necessary? I was OK with the string change, it's funny.

File: screenpy_requests/questions/body_of_the_last_response.py (hunk 2 above)
Comment (perrygoy) [reply screenpy_requests/questions/body_of_the_last_response.py]: Yeah, the main point of moving project code for type hinting into the `if TYPE_CHECKING` block is to reduce the chance of circular imports caused by type hints, and importing from `typing` will never cause that. 

But i kinda like just moving _all_ the type hinting stuff into that block, since we're using it anyway. I dunno, personal preference.

File: tests/test_actions.py (hunk 4 above)
Comment (bandophahita) [reply tests/test_actions.py]: Oops.  I didn't need to leave the annotation in.  
I was trying to fix a problem with mypy not liking `assert test_method in SendTESTMethod.__doc__`  because `__doc__` can be `str | None` and  `str in None` is invalid.

Once I added the assertion `assert SendTESTMethod.__doc__ is not None` mypy calmed down.   I just forgot to take the annotation off.

**Comment by bandophahita** [comment]
I updated the ruff config to push as many imports into TYPE_CHECKING as it can.

File: screenpy_requests/actions/add_header.py [hunk 5]
```diff
@@ -58,16 +61,15 @@ def perform_as(self, the_actor: Actor) -> None:
...
         session.headers.update(self.headers)
 
-    def __init__(
-        self, *header_pairs: Union[str, Iterable], **header_kwargs: str
-    ) -> None:
+    def __init__(self, *header_pairs: str | Iterable, **header_kwargs: str) -> None:
         self.headers = {}
         if len(header_pairs) == 1:
-            self.headers = dict(header_pairs[0])  # type: ignore
+            self.headers = dict(cast(Iterable, header_pairs[0]))
```
Comment (perrygoy) [inline screenpy_requests/actions/add_header.py:67]: Trust us `mypy` we know what we're talkin' about here.

File: screenpy_requests/questions/body_of_the_last_response.py [hunk 6]
```diff
@@ -71,14 +72,14 @@ def answered_by(self, the_actor: Actor) -> dict | str:
...
             response = responses[-1].json()
             for part in self.body_parts:
                 response = response[part]
-            return response
         except JSONDecodeError:
             response = responses[-1].text
             for part in self.body_parts:
                 response = response[part]
-            return response
+        return response
```
Comment (perrygoy) [inline screenpy_requests/questions/body_of_the_last_response.py:85]: Man, i really should have noticed this one. Hahaha. Thanks, linting!

File: setup.py
Comment (perrygoy) [inline setup.py:1]: I removed `setup.py` from `screenpy_playwright` and everything worked... maybe it's time to delete this one in the "Project standardization" PR?

**Review by perrygoy — APPROVED** [review]
A few comments of support, looks good! Thanks for getting this in @bandophahita!
